enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,307 @@
+to diagnose and fix issues.
+
+### User Stories
+
+* As a MicroShift administrator, I want to be able to get a report about current
+  cluster status.
+* As a MicroShift developer, I want to be able to diagnose issues with a detailed
+  debug report.
+
+### Goals
+* Provide a single tool to automate the report creation.
+
+### Non-Goals
+N/A
+
+## Proposal
+While `must-gather` tool is a widely known and used for debugging clusters, it
+may not be the most suitable for MicroShift use cases, and requires deep
+modifications. The existing `must-gather` implementation:
+1. Needs many OpenShift APIs present, like `clusteroperators`.
+1. Expects certain OpenShift components to be present, like
+machine-config-daemon.
+1. Runs commands within a pod, which means pulling another container over
+network (or pre-caching it).
+1. We want the debugging workflow for the product to be consistent, regardless
+of what is running on it and sos pulls info from many components not present in
+MicroShift.
+
+Some of these reasons are a no-go for MicroShift because of constrained
+environments. Some others mean there should be significant changes to
+`must-gather` to have MicroShift awareness.
+
+There is another tool for gathering info from a running RHEL system, and that
+is [`sos`](https://sos.readthedocs.org/). This tool is present in RHEL repos
+and is capable of generating reports for many components and applications in
+the system by using plugins. By executing `sos report` it will generate a
+detailed report with all the available information it can extract.
+
+Plugins have an enablement check or may be invoked manually when executing
+the command. A [plugin](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+is a python script that we include as part of `sos`.
+
+There is a [plugin for OpenShift](https://github.com/sosreport/sos/blob/main/sos/report/plugins/openshift.py)
+already that almost fits the needs of this EP. MicroShift does not share the
+same resources, APIs and namespaces/names that the plugin assumes, as it was
+made for OpenShift clusters. However, much of the needs covered in the plugin
+are the same, so the contents of the plugin may be used too.
+
+A simple `sos report` will yield much of the information needed to debug a
+MicroShift deployment (the list of active plugins can be obtained by running
+`sos report -l`). However, there are some specifics that we need to add to a
+new plugin:
+1. Control plane component logs: apiserver, etcd, kubelet, etc. (these are
+retrieved by syslog, meaning they are mixed with other system logs).
+1. Yaml output for relevant manifests from API server.
+1. Logs for every container. These are already retrieved by CRIO plugin. May
+need easier indexing.
+1. MicroShift's configuration and version. This [enhancement](https://github.com/openshift/enhancements/pull/1304)
+might add additional information to gather: plugins, manifests, etc.
```

Comment:
Note that sosreport not only collects the OpenShift data, but platform data as well! There are another 50+ plugins which are executed. The sosreport can run in different profile, where only necessary plugins are executed.

Reply:
We resolved to use a configuration file that MicroShift provides upon install with the required plugins. Then it is easy to run `sos report --config-file /etc/microshift/sos.conf`, for example.